python312Packages.huggingface-hub: 0.27.1 -> 0.28.1, python312Packages.transformers: 4.48.1 -> 4.48.2

GaetanLepage opened the pull request:
## Things done

Changelog: https://github.com/huggingface/huggingface_hub/releases/tag/v0.28.1

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

GaetanLepage committed:
python312Packages.huggingface-hub: 0.27.1 -> 0.28.1

Diff: https://github.com/huggingface/huggingface_hub/compare/refs/tags/v0.27.1...v0.28.1

Changelog: https://github.com/huggingface/huggingface_hub/releases/tag/v0.28.1

sarahec commented:
I'm running `nixpkgs-review` and so far so good. I'm curious about the huge fanout and wondering what package is the culprit. Surely all the downstream packages aren't importing this directly?

GaetanLepage committed:
python312Packages.transformers: 4.48.1 -> 4.48.2

Diff: https://github.com/huggingface/transformers/compare/refs/tags/v4.48.1...v4.48.2

Changelog: https://github.com/huggingface/transformers/releases/tag/v4.48.2

sarahec commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 378039`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 14 packages marked as broken and skipped:</summary>
  <ul>
    <li>dbx</li>
    <li>dbx.dist</li>
    <li>khoj</li>
    <li>khoj.dist</li>
    <li>python312Packages.haystack-ai</li>
    <li>python312Packages.haystack-ai.dist</li>
    <li>python312Packages.mamba-ssm</li>
    <li>python312Packages.mamba-ssm.dist</li>
    <li>python312Packages.vllm</li>
    <li>python312Packages.vllm.dist</li>
    <li>python313Packages.haystack-ai</li>
    <li>python313Packages.haystack-ai.dist</li>
    <li>python313Packages.mamba-ssm</li>
    <li>python313Packages.mamba-ssm.dist</li>
  </ul>
</details>
<details>
  <summary>:x: 64 packages failed to build:</summary>
  <ul>
    <li>private-gpt</li>
    <li>private-gpt.dist</li>
    <li>python312Packages.docling-ibm-models</li>
    <li>python312Packages.docling-ibm-models.dist</li>
    <li>python312Packages.fastembed</li>
    <li>python312Packages.fastembed.dist</li>
    <li>python312Packages.kserve</li>
    <li>python312Packages.kserve.dist</li>
    <li>python312Packages.private-gpt</li>
    <li>python312Packages.private-gpt.dist</li>
    <li>python313Packages.cohere</li>
    <li>python313Packages.cohere.dist</li>
    <li>python313Packages.databricks-sdk</li>
    <li>python313Packages.databricks-sdk.dist</li>
    <li>python313Packages.effdet</li>
    <li>python313Packages.effdet.dist</li>
    <li>python313Packages.fastembed</li>
    <li>python313Packages.fastembed.dist</li>
    <li>python313Packages.gradio</li>
    <li>python313Packages.gradio-client</li>
    <li>python313Packages.gradio-client.dist</li>
    <li>python313Packages.gradio-pdf</li>
    <li>python313Packages.gradio-pdf.dist</li>
    <li>python313Packages.gradio.dist</li>
    <li>python313Packages.instructor</li>
    <li>python313Packages.instructor.dist</li>
    <li>python313Packages.langchain</li>
    <li>python313Packages.langchain-aws</li>
    <li>python313Packages.langchain-aws.dist</li>
    <li>python313Packages.langchain-azure-dynamic-sessions</li>
    <li>python313Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python313Packages.langchain-community</li>
    <li>python313Packages.langchain-community.dist</li>
    <li>python313Packages.langchain-core</li>
    <li>python313Packages.langchain-core.dist</li>
    <li>python313Packages.langchain-huggingface</li>
    <li>python313Packages.langchain-huggingface.dist</li>
    <li>python313Packages.langchain-mongodb</li>
    <li>python313Packages.langchain-mongodb.dist</li>
    <li>python313Packages.langchain-ollama</li>
    <li>python313Packages.langchain-ollama.dist</li>
    <li>python313Packages.langchain-openai</li>
    <li>python313Packages.langchain-openai.dist</li>
    <li>python313Packages.langchain-tests</li>
    <li>python313Packages.langchain-tests.dist</li>
    <li>python313Packages.langchain-text-splitters</li>
    <li>python313Packages.langchain-text-splitters.dist</li>
    <li>python313Packages.langchain.dist</li>
    <li>python313Packages.langgraph-checkpoint</li>
    <li>python313Packages.langgraph-checkpoint-duckdb</li>
    <li>python313Packages.langgraph-checkpoint-duckdb.dist</li>
    <li>python313Packages.langgraph-checkpoint-sqlite</li>
    <li>python313Packages.langgraph-checkpoint-sqlite.dist</li>
    <li>python313Packages.langgraph-checkpoint.dist</li>
    <li>python313Packages.langsmith</li>
    <li>python313Packages.langsmith.dist</li>
    <li>python313Packages.llmx</li>
    <li>python313Packages.llmx.dist</li>
    <li>python313Packages.mlflow</li>
    <li>python313Packages.mlflow.dist</li>
    <li>python313Packages.sagemaker-mlflow</li>
    <li>python313Packages.sagemaker-mlflow.dist</li>
    <li>python313Packages.unstructured-inference</li>
    <li>python313Packages.unstructured-inference.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 261 packages built:</summary>
  <ul>
    <li>aider-chat (python312Packages.aider-chat)</li>
    <li>aider-chat.dist (python312Packages.aider-chat.dist)</li>
    <li>datatrove</li>
    <li>datatrove.dist</li>
    <li>exo</li>
    <li>exo.dist</li>
    <li>home-assistant-component-tests.anthropic</li>
    <li>immich-machine-learning</li>
    <li>immich-machine-learning.dist</li>
    <li>libretranslate (python312Packages.libretranslate)</li>
    <li>libretranslate.dist (python312Packages.libretranslate.dist)</li>
    <li>minari (python312Packages.minari)</li>
    <li>minari.dist (python312Packages.minari.dist)</li>
    <li>mlflow-server</li>
    <li>mlflow-server.dist</li>
    <li>mokuro</li>
    <li>mokuro.dist</li>
    <li>open-interpreter (python312Packages.open-interpreter)</li>
    <li>open-interpreter.dist (python312Packages.open-interpreter.dist)</li>
    <li>open-webui</li>
    <li>open-webui.dist</li>
    <li>pentestgpt</li>
    <li>pentestgpt.dist</li>
    <li>python312Packages.accelerate</li>
    <li>python312Packages.accelerate.dist</li>
    <li>python312Packages.anthropic</li>
    <li>python312Packages.anthropic.dist</li>
    <li>python312Packages.argos-translate-files</li>
    <li>python312Packages.argos-translate-files.dist</li>
    <li>python312Packages.argostranslate</li>
    <li>python312Packages.argostranslate.dist</li>
    <li>python312Packages.chromadb</li>
    <li>python312Packages.chromadb.dist</li>
    <li>python312Packages.cleanlab</li>
    <li>python312Packages.cleanlab.dist</li>
    <li>python312Packages.cleanvision</li>
    <li>python312Packages.cleanvision.dist</li>
    <li>python312Packages.cohere</li>
    <li>python312Packages.cohere.dist</li>
    <li>python312Packages.colbert-ai</li>
    <li>python312Packages.colbert-ai.dist</li>
    <li>python312Packages.ctranslate2</li>
    <li>python312Packages.ctranslate2.dist</li>
    <li>python312Packages.dalle-mini</li>
    <li>python312Packages.dalle-mini.dist</li>
    <li>python312Packages.databricks-sdk</li>
    <li>python312Packages.databricks-sdk.dist</li>
    <li>python312Packages.datasets</li>
    <li>python312Packages.datasets.dist</li>
    <li>python312Packages.diffusers</li>
    <li>python312Packages.diffusers.dist</li>
    <li>python312Packages.effdet</li>
    <li>python312Packages.effdet.dist</li>
    <li>python312Packages.evaluate</li>
    <li>python312Packages.evaluate.dist</li>
    <li>python312Packages.faster-whisper</li>
    <li>python312Packages.faster-whisper.dist</li>
    <li>python312Packages.gradio</li>
    <li>python312Packages.gradio-client</li>
    <li>python312Packages.gradio-client.dist</li>
    <li>python312Packages.gradio-pdf</li>
    <li>python312Packages.gradio-pdf.dist</li>
    <li>python312Packages.gradio.dist</li>
    <li>python312Packages.guidance</li>
    <li>python312Packages.guidance.dist</li>
    <li>python312Packages.huggingface-hub</li>
    <li>python312Packages.huggingface-hub.dist</li>
    <li>python312Packages.instructor</li>
    <li>python312Packages.instructor.dist</li>
    <li>python312Packages.k-diffusion</li>
    <li>python312Packages.k-diffusion.dist</li>
    <li>python312Packages.langchain</li>
    <li>python312Packages.langchain-aws</li>
    <li>python312Packages.langchain-aws.dist</li>
    <li>python312Packages.langchain-azure-dynamic-sessions</li>
    <li>python312Packages.langchain-azure-dynamic-sessions.dist</li>
    <li>python312Packages.langchain-chroma</li>
    <li>python312Packages.langchain-chroma.dist</li>
    <li>python312Packages.langchain-community</li>
    <li>python312Packages.langchain-community.dist</li>
    <li>python312Packages.langchain-core</li>
    <li>python312Packages.langchain-core.dist</li>
    <li>python312Packages.langchain-huggingface</li>
    <li>python312Packages.langchain-huggingface.dist</li>
    <li>python312Packages.langchain-mongodb</li>
    <li>python312Packages.langchain-mongodb.dist</li>
    <li>python312Packages.langchain-ollama</li>
    <li>python312Packages.langchain-ollama.dist</li>
    <li>python312Packages.langchain-openai</li>
    <li>python312Packages.langchain-openai.dist</li>
    <li>python312Packages.langchain-tests</li>
    <li>python312Packages.langchain-tests.dist</li>
    <li>python312Packages.langchain-text-splitters</li>
    <li>python312Packages.langchain-text-splitters.dist</li>
    <li>python312Packages.langchain.dist</li>
    <li>python312Packages.langgraph</li>
    <li>python312Packages.langgraph-checkpoint</li>
    <li>python312Packages.langgraph-checkpoint-duckdb</li>
    <li>python312Packages.langgraph-checkpoint-duckdb.dist</li>
    <li>python312Packages.langgraph-checkpoint-postgres</li>
    <li>python312Packages.langgraph-checkpoint-postgres.dist</li>
    <li>python312Packages.langgraph-checkpoint-sqlite</li>
    <li>python312Packages.langgraph-checkpoint-sqlite.dist</li>
    <li>python312Packages.langgraph-checkpoint.dist</li>
    <li>python312Packages.langgraph.dist</li>
    <li>python312Packages.langsmith</li>
    <li>python312Packages.langsmith.dist</li>
    <li>python312Packages.litellm</li>
    <li>python312Packages.litellm.dist</li>
    <li>python312Packages.llama-cpp-python</li>
    <li>python312Packages.llama-cpp-python.dist</li>
    <li>python312Packages.llama-index</li>
    <li>python312Packages.llama-index-cli</li>
    <li>python312Packages.llama-index-cli.dist</li>
    <li>python312Packages.llama-index-embeddings-huggingface</li>
    <li>python312Packages.llama-index-embeddings-huggingface.dist</li>
    <li>python312Packages.llama-index-llms-openai-like</li>
    <li>python312Packages.llama-index-llms-openai-like.dist</li>
    <li>python312Packages.llama-index-vector-stores-chroma</li>
    <li>python312Packages.llama-index-vector-stores-chroma.dist</li>
    <li>python312Packages.llama-index.dist</li>
    <li>python312Packages.llm-gguf</li>
    <li>python312Packages.llm-gguf.dist</li>
    <li>python312Packages.llmx</li>
    <li>python312Packages.llmx.dist</li>
    <li>python312Packages.manga-ocr</li>
    <li>python312Packages.manga-ocr.dist</li>
    <li>python312Packages.manifest-ml</li>
    <li>python312Packages.manifest-ml.dist</li>
    <li>python312Packages.mlflow</li>
    <li>python312Packages.mlflow.dist</li>
    <li>python312Packages.mmcv</li>
    <li>python312Packages.mmcv.dist</li>
    <li>python312Packages.mmengine</li>
    <li>python312Packages.mmengine.dist</li>
    <li>python312Packages.open-clip-torch</li>
    <li>python312Packages.open-clip-torch.dist</li>
    <li>python312Packages.optimum</li>
    <li>python312Packages.optimum.dist</li>
    <li>python312Packages.outlines</li>
    <li>python312Packages.outlines-core</li>
    <li>python312Packages.outlines-core.dist</li>
    <li>python312Packages.outlines.dist</li>
    <li>python312Packages.peft</li>
    <li>python312Packages.peft.dist</li>
    <li>python312Packages.pyannote-audio</li>
    <li>python312Packages.pyannote-audio.dist</li>
    <li>python312Packages.sagemaker-mlflow</li>
    <li>python312Packages.sagemaker-mlflow.dist</li>
    <li>python312Packages.sentence-transformers</li>
    <li>python312Packages.sentence-transformers.dist</li>
    <li>python312Packages.skops</li>
    <li>python312Packages.skops.dist</li>
    <li>python312Packages.spacy-transformers</li>
    <li>python312Packages.spacy-transformers.dist</li>
    <li>python312Packages.speechbrain</li>
    <li>python312Packages.speechbrain.dist</li>
    <li>python312Packages.stanza</li>
    <li>python312Packages.stanza.dist</li>
    <li>python312Packages.timm</li>
    <li>python312Packages.timm.dist</li>
    <li>python312Packages.tinygrad</li>
    <li>python312Packages.tinygrad.dist</li>
    <li>python312Packages.tokenizers</li>
    <li>python312Packages.tokenizers.dist</li>
    <li>python312Packages.torchinfo</li>
    <li>python312Packages.torchinfo.dist</li>
    <li>python312Packages.transformers</li>
    <li>python312Packages.transformers.dist</li>
    <li>python312Packages.translatehtml</li>
    <li>python312Packages.translatehtml.dist</li>
    <li>python312Packages.txtai</li>
    <li>python312Packages.txtai.dist</li>
    <li>python312Packages.unstructured-inference</li>
    <li>python312Packages.unstructured-inference.dist</li>
    <li>python312Packages.vqgan-jax</li>
    <li>python312Packages.vqgan-jax.dist</li>
    <li>python313Packages.accelerate</li>
    <li>python313Packages.accelerate.dist</li>
    <li>python313Packages.anthropic</li>
    <li>python313Packages.anthropic.dist</li>
    <li>python313Packages.argos-translate-files</li>
    <li>python313Packages.argos-translate-files.dist</li>
    <li>python313Packages.argostranslate</li>
    <li>python313Packages.argostranslate.dist</li>
    <li>python313Packages.cleanvision</li>
    <li>python313Packages.cleanvision.dist</li>
    <li>python313Packages.colbert-ai</li>
    <li>python313Packages.colbert-ai.dist</li>
    <li>python313Packages.ctranslate2</li>
    <li>python313Packages.ctranslate2.dist</li>
    <li>python313Packages.datasets</li>
    <li>python313Packages.datasets.dist</li>
    <li>python313Packages.diffusers</li>
    <li>python313Packages.diffusers.dist</li>
    <li>python313Packages.docling-ibm-models</li>
    <li>python313Packages.docling-ibm-models.dist</li>
    <li>python313Packages.evaluate</li>
    <li>python313Packages.evaluate.dist</li>
    <li>python313Packages.faster-whisper</li>
    <li>python313Packages.faster-whisper.dist</li>
    <li>python313Packages.guidance</li>
    <li>python313Packages.guidance.dist</li>
    <li>python313Packages.huggingface-hub</li>
    <li>python313Packages.huggingface-hub.dist</li>
    <li>python313Packages.k-diffusion</li>
    <li>python313Packages.k-diffusion.dist</li>
    <li>python313Packages.libretranslate</li>
    <li>python313Packages.libretranslate.dist</li>
    <li>python313Packages.litellm</li>
    <li>python313Packages.litellm.dist</li>
    <li>python313Packages.llama-cpp-python</li>
    <li>python313Packages.llama-cpp-python.dist</li>
    <li>python313Packages.llm-gguf</li>
    <li>python313Packages.llm-gguf.dist</li>
    <li>python313Packages.manifest-ml</li>
    <li>python313Packages.manifest-ml.dist</li>
    <li>python313Packages.open-clip-torch</li>
    <li>python313Packages.open-clip-torch.dist</li>
    <li>python313Packages.open-interpreter</li>
    <li>python313Packages.open-interpreter.dist</li>
    <li>python313Packages.optimum</li>
    <li>python313Packages.optimum.dist</li>
    <li>python313Packages.outlines</li>
    <li>python313Packages.outlines-core</li>
    <li>python313Packages.outlines-core.dist</li>
    <li>python313Packages.outlines.dist</li>
    <li>python313Packages.peft</li>
    <li>python313Packages.peft.dist</li>
    <li>python313Packages.sentence-transformers</li>
    <li>python313Packages.sentence-transformers.dist</li>
    <li>python313Packages.skops</li>
    <li>python313Packages.skops.dist</li>
    <li>python313Packages.speechbrain</li>
    <li>python313Packages.speechbrain.dist</li>
    <li>python313Packages.stanza</li>
    <li>python313Packages.stanza.dist</li>
    <li>python313Packages.timm</li>
    <li>python313Packages.timm.dist</li>
    <li>python313Packages.tokenizers</li>
    <li>python313Packages.tokenizers.dist</li>
    <li>python313Packages.torchinfo</li>
    <li>python313Packages.torchinfo.dist</li>
    <li>python313Packages.transformers</li>
    <li>python313Packages.transformers.dist</li>
    <li>python313Packages.translatehtml</li>
    <li>python313Packages.translatehtml.dist</li>
    <li>rclip</li>
    <li>rclip.dist</li>
    <li>seagoat</li>
    <li>seagoat.dist</li>
    <li>shell-gpt</li>
    <li>shell-gpt.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
    <li>unstructured-api</li>
    <li>vimPlugins.aider-nvim</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
    <li>wyoming-faster-whisper</li>
    <li>wyoming-faster-whisper.dist</li>
  </ul>
</details>

sarahec commented:
> Thanks @sarahec. Although I pushed the bump of `transformers` too.

I saw that and restarted `nixpkgs-review` to account for it.

Note: Most Python 3.13 failures are due to #378154 and fixed in #378156
